### PR TITLE
New preset implementation and test

### PR DESCRIPTION
**Pull request by lizlouise1335**
Implemented memory_optimization preset, updated set() to take str:Any dictionaries instead of just str:bool to accommodate new preset, implemented memory_optimization preset test

### CODE REVIEW

File: dataprofiler/profilers/profiler_options.py [hunk 1]
```diff
@@ -1583,6 +1587,27 @@ def _numeric_stats_disabled_presets(self) -> None:
         self.set({"*.float.is_numeric_stats_enabled": False})
         self.set({"structured_options.text.is_numeric_stats_enabled": False})
 
+    def _memory_optimization_presets(self) -> None:  # split into more dicts
```
Comment (taylorfturner) [inline dataprofiler/profilers/profiler_options.py:1590]: i don't think this comment is needed.... 

```suggestion
    def _memory_optimization_presets(self) -> None: 
```

File: dataprofiler/profilers/profiler_options.py [hunk 2]
```diff
@@ -1620,7 +1645,7 @@ def _validate_helper(self, variable_path: str = "ProfilerOptions") -> list[str]:
 
         return errors
 
-    def set(self, options: dict[str, bool]) -> None:
+    def set(self, options: dict[str, Any]) -> None:
```
Comment (taylorfturner) [inline dataprofiler/profilers/profiler_options.py:1648]: nice

File: dataprofiler/profilers/profiler_options.py [hunk 3]
```diff
@@ -1570,6 +1572,8 @@ def __init__(self, presets: str = None) -> None:
                 self._data_types_presets()
             elif self.presets == "numeric_stats_disabled":
                 self._numeric_stats_disabled_presets()
+            elif self.presets == "memory_optimization":
+                self._memory_optimization_presets()
```
Comment (taylorfturner) [inline dataprofiler/profilers/profiler_options.py:1576]: let's add some validation in `_validate_helper` for the universe of `presets` options that are valid. It should fail if not valid preset option

**Review by taylorfturner — CHANGES_REQUESTED** [review]

File: dataprofiler/profilers/profiler_options.py [hunk 4]
```diff
@@ -1583,6 +1587,27 @@ def _numeric_stats_disabled_presets(self) -> None:
...
+                "structured_options.row_statistics.is_enabled": False,
+                "structured_options.multiprocess.is_enabled": False,
+                "structured_options.data_labeler.is_enabled": False,
+                "structured_options.datetime.is_enabled": False,
+                "structured_options.order.is_enabled": False,
+                "structured_options.chi2_homogeneity.is_enabled": False,
+                "structured_options.null_replication_metrics.is_enabled": False,
+                "unstructured_options.data_labeler.is_enabled": False,
+                """structured_options.category.
+                max_sample_size_to_check_stop_condition""": 250,
```
Comment (taylorfturner) [inline dataprofiler/profilers/profiler_options.py:1602]: yeah the test on python3.10 doesn't like how this is formatted... 

```
                )
E               AttributeError: type object 'structured_options.category' has no attribute '
E                               max_sample_size_to_check_stop_condition'
```

**Review by taylorfturner — APPROVED** [review]

File: dataprofiler/profilers/profiler_options.py [hunk 5]
```diff
@@ -1583,6 +1589,25 @@ def _numeric_stats_disabled_presets(self) -> None:
         self.set({"*.float.is_numeric_stats_enabled": False})
         self.set({"structured_options.text.is_numeric_stats_enabled": False})
 
+    def _memory_optimization_presets(self) -> None:
+        self.set({"structured_options.row_statistics.is_enabled": False})
```
Comment (micdavis) [inline dataprofiler/profilers/profiler_options.py:1593]: should `row_statistics` be enabled with HLL @taylorfturner

File: dataprofiler/profilers/profiler_options.py [hunk 6]
```diff
@@ -1583,6 +1589,25 @@ def _numeric_stats_disabled_presets(self) -> None:
...
+        self.set({"structured_options.multiprocess.is_enabled": False})
+        self.set({"structured_options.data_labeler.is_enabled": False})
+        self.set({"structured_options.datetime.is_enabled": False})
+        self.set({"structured_options.order.is_enabled": False})
```
Comment (micdavis) [inline dataprofiler/profilers/profiler_options.py:1597]: curious why we are specifically disabling these two. Are they memory intensive?

**Comment by micdavis** [comment]
Functionally, it looks good, just curious about the "why" on some of the options that are disabled in the presets

File: dataprofiler/profilers/profiler_options.py (hunk 5 above)
Comment (taylorfturner) [reply dataprofiler/profilers/profiler_options.py:1593]: that's more TBD IMO

**Review by micdavis — APPROVED** [review]

**Pull request merged**